core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -113,7 +116,89 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
                             entityNames: Seq[String],
                             linkExistingEntities: Boolean,
                             parentContext: RawlsRequestContext
-  ): Future[EntityCopyResponse] = ???
+  ): Future[EntityCopyResponse] =
+    checkAndCopyEntities(
+      sourceWorkspaceContext,
+      destWorkspaceContext,
+      entityType,
+      entityNames,
+      linkExistingEntities,
+      parentContext
+    )
+
+  def checkAndCopyEntities(sourceWorkspaceContext: Workspace,
+                           destWorkspaceContext: Workspace,
+                           entityType: String,
+                           entityNames: Seq[String],
+                           linkExistingEntities: Boolean,
+                           parentContext: RawlsRequestContext
+  ): Future[EntityCopyResponse] = {
+
+    def getSoftConflicts(paths: Set[EntityPointer]): Future[Set[EntityPointer]] =
+      // return the entities already present in the destination workspace
+      repository.dataSource.inTransaction { _ =>
```

Comment:
Does it make sense to have 1 transaction for this whole function instead of calling `inTransaction` 4 times? This would make immune from race conditions where data is updated between transactions but perhaps introduce locking issues.